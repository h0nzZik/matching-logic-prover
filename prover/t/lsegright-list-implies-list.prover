claim \implies( \and( listSegmentRight( variable("H") { ArrayIntInt }
                                      , variable("X") { Int }
                                      , variable("Y") { Int }
                                      , variable("G1") { Set }
                                      , .Patterns)
                    , list( variable("H") { ArrayIntInt }
                          , variable("Y") { Int }
                          , variable("G2") { Set }
                          , .Patterns
                          )
                    , disjoint(variable("G1") { Set }, variable("G2") { Set })
                    , \equals(variable("F") { Set }, union(variable("G1") { Set }, variable("G2") { Set }))
                    , .Patterns )
              , \and( list( variable("H") { ArrayIntInt }
                          , variable("X") { Int }
                          , variable("F") { Set }
                          , .Patterns)
                    , .Patterns )
              )

strategy kt # head(listSegmentRight) # freshPositions(1)
       ; ( ( simplify ; direct-proof )
         | ( right-unfold-Nth(0, 1) ; simplify ; direct-proof )
         )
