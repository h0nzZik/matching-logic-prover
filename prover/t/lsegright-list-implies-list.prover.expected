<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( listSegmentRight ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" ) { Int } , variable ( "G1" ) { Set } , .Patterns ) , list ( variable ( "H" ) { ArrayIntInt } , variable ( "Y" ) { Int } , variable ( "G2" ) { Set } , .Patterns ) , disjoint ( variable ( "G1" ) { Set } , variable ( "G2" ) { Set } ) , \equals ( variable ( "F" ) { Set } , union ( variable ( "G1" ) { Set } , variable ( "G2" ) { Set } ) ) , .Patterns ) , \and ( list ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "F" ) { Set } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
